base/server/src/org/dogtagpki/server/rest/SecurityDomainHostService.java

```diff
@@ -0,0 +1,177 @@
+        logger.debug("SecurityDomainService: domain: " + domainInfo.getName());
+
+        for (SecurityDomainSubsystem subsystem : domainInfo.getSubsystems().values()) {
+            for (SecurityDomainHost host : subsystem.getHosts().values()) {
+
+                logger.debug("SecurityDomainService: - " + host.getId());
+
+                if (host.getId().equals(hostID)) {
+                    logger.debug("SecurityDomainService: Found security domain host " + hostID);
+                    return createOKResponse(host);
+                }
+            }
+        }
+
+        throw new ResourceNotFoundException("Security domain host not found: " + hostID);
+    }
+
+    @Override
+    public Response addHost(SecurityDomainHost host) throws Exception {
+
+        String hostID = host.getId();
+        logger.info("SecurityDomainService: Adding security domain host \"" + hostID + "\"");
+
+        // Host ID: <type> <hostname> <port>
+        Pattern pattern = Pattern.compile("^(\\S+) (\\S+) (\\d+)$");
+        Matcher matcher = pattern.matcher(hostID);
+
+        if (!matcher.find()) {
+            throw new BadRequestException("Invalid security domain host: " + hostID);
+        }
+
+        String type = matcher.group(1);
+        logger.debug("SecurityDomainService: type: " + type);
+
+        String hostname = matcher.group(2);
+        logger.debug("SecurityDomainService: hostname: " + hostname);
+
+        String port = matcher.group(3);
+        logger.debug("SecurityDomainService: port: " + port);
+
+        String unsecurePort = host.getPort();
+        logger.debug("SecurityDomainService: unsecure port: " + unsecurePort);
+
+        String eeCAPort = host.getSecureEEClientAuthPort();
+        logger.debug("SecurityDomainService: EE port: " + eeCAPort);
+
+        String adminSecurePort = host.getSecureAdminPort();
+        logger.debug("SecurityDomainService: admin port: " + adminSecurePort);
+
+        String agentSecurePort = host.getSecureAgentPort();
+        logger.debug("SecurityDomainService: agent port: " + agentSecurePort);
+
+        String domainManager = host.getDomainManager();
+        logger.debug("SecurityDomainService: domain manager: " + domainManager);
+
+        String clone = host.getClone();
+        logger.debug("SecurityDomainService: clone: " + clone);
+
+        SecurityDomainProcessor processor = new SecurityDomainProcessor(getLocale(headers));
+        String status = processor.addHost(
+                hostID,
+                type,
+                hostname,
+                port,
+                unsecurePort,
+                eeCAPort,
+                adminSecurePort,
+                agentSecurePort,
+                domainManager,
+                clone);
+        logger.debug("SecurityDomainService: status: " + status);
+
+        if (!SecurityDomainProcessor.SUCCESS.equals(status)) {
+            throw new PKIException("Unable to remove security domain host: " + hostID);
```

Comment:
Typo: "Unable to **add** security domain host"?

Reply:
Good catch! Fixed.